deploy/clickhouse-keeper/clickhouse-keeper-1-node-256M-for-test-only.yaml

```diff
@@ -0,0 +1,246 @@
+---
+# Fake Service to drop-in replacement in tests
+apiVersion: v1
+kind: Service
+metadata:
+  # DNS would be like zookeeper.namespace.svc
+  name: zookeeper
+  labels:
+    app: zookeeper
+spec:
+  ports:
+    - port: 2181
+      name: client
+    - port: 7000
+      name: prometheus
+  selector:
+    app: clickhouse-keeper
+    what: node
+---
+# Setup Service to provide access to ClickHouse keeper for clients
+apiVersion: v1
+kind: Service
+metadata:
+  # DNS would be like clickhouse-keeper.namespace.svc
+  name: clickhouse-keeper
+  labels:
+    app: clickhouse-keeper
+spec:
+  ports:
+    - port: 2181
+      name: client
+    - port: 7000
+      name: prometheus
+  selector:
+    app: clickhouse-keeper
+    what: node
+---
+# Setup Headless Service for StatefulSet
+apiVersion: v1
+kind: Service
+metadata:
+  # DNS would be like keeper-0.clickhouse-keepers.namespace.svc
+  name: clickhouse-keepers
+  labels:
+    app: clickhouse-keeper
+spec:
+  ports:
+    - port: 9234
+      name: raft
+  clusterIP: None
+  selector:
+    app: clickhouse-keeper
+    what: node
+---
+# Setup max number of unavailable pods in StatefulSet
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: clickhouse-keeper-pod-disruption-budget
+spec:
+  selector:
+    matchLabels:
+      app: clickhouse-keeper
+  maxUnavailable: 1
+---
+# Setup ClickHouse Keeper settings
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: clickhouse-keeper-settings
+data:
+  keeper_config.xml: |
+    <yandex>
+        <include_from>/tmp/clickhouse-keeper/config.d/generated-keeper-settings.xml</include_from>
+        <logger>
+            <level>trace</level>
+            <console>true</console>
+        </logger>
+        <listen_host>0.0.0.0</listen_host>
+        <keeper_server incl="keeper_server">
+            <path>/var/lib/clickhouse-keeper</path>
+            <tcp_port>2181</tcp_port>
+            <coordination_settings>
+                <!-- <raft_logs_level>trace</raft_logs_level> -->
+                <raft_logs_level>debug</raft_logs_level>
+            </coordination_settings>
+        </keeper_server>
+    </yandex>
+
+---
+# Setup ClickHouse Keeper StatefulSet
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  # nodes would be named as clickhouse-keeper-0, clickhouse-keeper-1, clickhouse-keeper-2
+  name: clickhouse-keeper
+  labels:
+    app: clickhouse-keeper
+spec:
+  selector:
+    matchLabels:
+      app: clickhouse-keeper
+  serviceName: clickhouse-keepers
+  replicas: 1
+  updateStrategy:
+    type: RollingUpdate
```

Comment:
Actually to reduce the number of leader shifts the best strategy is restarting the leader node last (same as for zookeeper). But i don't know if it can be programmed somehow in k8. (simple option should also work)

Reply:
We have `kind: PodDisruptionBudget` to avoid situation when more than one pod will restart on normal cases